runner/src/mill/runner/MillMain.scala

```diff
@@ -160,6 +162,25 @@ object MillMain {
 
             // special BSP mode, in which we spawn a server and register the current evaluator when-ever we start to eval a dedicated command
             val bspMode = config.bsp.value && config.leftoverArgs.value.isEmpty
+            val bspInstallModeArgsOpt =
+              Option.when(
+                !config.bsp.value &&
+                  config.leftoverArgs.value.headOption.contains("mill.bsp.BSP/install")
+              ) {
+                val expectedSelector = Seq((
+                  Some(Segments.labels("mill", "bsp", "BSP")),
+                  Some(Segments.labels("install"))
+                ))
+
+                for {
+                  parsed <- ParseArgs(config.leftoverArgs.value, SelectMode.Separated).toOption
+                  case Seq((`expectedSelector`, args)) <- Option(parsed)
```

Comment:
This nested code to try and detect the `install` command looks very convoluted. Would just checking `config.leftoverArgs.value.headOption == Some("mill.bsp.BSP/install")` be enough?

Reply:
I did that to parse any `--jobs` argument passed to it (whose value ends up being passed to the `./mill --bsp` command). I don't rely on it, but I don't know for other people out there.

Reply:
Seems like IntelliJ doesn't pass the flag resulting in the default of `--jobs 1` https://github.com/search?q=repo%3AJetBrains%2Fintellij-scala+%22mill.bsp.BSP%22&type=code, while Metals passes `--jobs 0` https://github.com/search?q=repo%3Ascalameta%2Fmetals%20%22mill.bsp.BSP%22&type=code

I think for now let's just ignore the `--jobs` config here to simplify our code, and either hardcode 1 or hardcode 0 (probably doesn't really matter).

Reply:
What is the situation in which this is needed at all?

Reply:
I guess this is the special handling needed to make `install` work even when the build does not work. If we keep it here, we should add some comment sufficiently explaining the situation.

I don't like this, tbh. If we're going to intercept the `mill.bsp.BSP/install` and just fake it, we might as well remove the original `Task.Command`. Ideally, we can use the same code in both situations, to avoid some config args to be accepted on one situations but not in the other. Of course, I'd like to keep the proper `--jobs` parsing, since this is currently the only way for a user to explicitly configure how the BSP server should run.

Reply:
The goal is in general moving the BSP logic out of `ExternalModule`s into the `mill-runner`, so really we should remove the `extends ExternalModule` from bsp/src/mill/bsp/BSP.scala and just make it normal helper methods

Reply:
Just removed the current `mill.bsp.BSP/install` command, and simplified the handling of `--jobs` in the manual handling of `mill.bsp.BSP/install` in Mill main